src/makefun/main.py

```diff
@@ -297,6 +341,28 @@ def _is_generator_func(func_impl):
         return isgeneratorfunction(func_impl)
 
 
+def _is_lambda_func_name(func_name):
+    """
+    Return True if func_name is the name of a lambda
+    :param func_name:
+    :return:
+    """
+    return func_name == (lambda: None).__code__.co_name
+
+
+def _is_valid_func_def_name(func_name):
+    """
+    Return True if func_name is valid in a function definition.
+    :param func_name:
+    :return:
+    """
+    if sys.version_info >= (3, 0):
+        is_identifier = str.isidentifier
+    else:
+        from makefun._main_legacy_py import is_identifier
+    return is_identifier(func_name) and not iskeyword(func_name)
```

Comment:
Let's add the `is_identifier` symbol at the top of the module so that it is checked only once

```suggestion
def _is_valid_func_def_name(func_name):
    """
    Return True if func_name is valid in a function definition.
    :param func_name:
    :return:
    """
    return is_identifier(func_name) and not iskeyword(func_name)
```